src/mmrelay/config.py

```
@@ -301,11 +355,24 @@ def get_explicit_credentials_path(config: Mapping[str, Any] | None) -> str | Non
     Returns:
         str | None: The configured credentials path string, or `None` if not set.
 
+    Precedence: MMRELAY_CREDENTIALS_PATH env var > top-level credentials_path > matrix.credentials_path.
+    All three sources are deprecated in favor of MMRELAY_HOME. Warnings are emitted once per process.
+
     Raises:
         InvalidCredentialsPathTypeError: If a found `credentials_path` value exists but is not a string.
     """
     env_path = os.getenv("MMRELAY_CREDENTIALS_PATH")
     if env_path:
+        global _legacy_path_override_warning_shown
```

Ruff (0.15.10)

[warning] Using the global statement to update `_legacy_path_override_warning_shown` is discouraged

(PLW0603)